synapse/util/async.py

```diff
@@ -227,7 +224,7 @@ def queue(self, key):
             # the context manager, but it needs to happen while we hold the
             # lock, and the context manager's exit code must be synchronous,
             # so actually this is the only sensible place.
-            yield run_on_reactor()
+            yield self._clock.sleep(0)
```

Comment:
can we make this do `sleep(0)` for now, and then replace it with `self._clock.sleep` in the PR which does the sleep replacement?

Reply:
Hmm, yeah. Don't think I meant to cherry pick this out. Fixed.